tutorials/v7/ntuple/ntpl015_streaming_vector.C

```diff
@@ -0,0 +1,161 @@
+/// \file
+/// \ingroup tutorial_ntuple
+///
+/// Example of a streaming vector: a special purpose container that reads large vectors piece-wise.
+///
+/// \macro_code
+///
+/// \date November 2024
+/// \author Peter van Gemmeren, the ROOT Team
+
+// NOTE: The RNTuple classes are experimental at this point.
+// Functionality and interface are still subject to changes.
+
+#include <ROOT/RNTupleModel.hxx>
+#include <ROOT/RNTupleReader.hxx>
+#include <ROOT/RNTupleUtil.hxx>
+#include <ROOT/RNTupleView.hxx>
+#include <ROOT/RNTupleWriter.hxx>
+
+#include <TRandom3.h>
+
+#include <cstdint>
+#include <iostream>
+#include <vector>
+#include <utility>
+
+using namespace ROOT::Experimental;
+
+constexpr char const *kFileName = "ntpl015_streaming_vector.root";
+constexpr char const *kNTupleName = "ntpl";
+constexpr char const *kFieldName = "LargeVector";
+constexpr unsigned int kNEvents = 10;
+constexpr unsigned int kVectorSize = 1000000;
+
+void CreateRNTuple()
+{
+   auto model = RNTupleModel::Create();
+   auto ptrLargeVector = model->MakeField<std::vector<std::uint32_t>>(kFieldName);
+   auto writer = RNTupleWriter::Recreate(std::move(model), kNTupleName, kFileName);
+
+   auto prng = std::make_unique<TRandom3>();
+   prng->SetSeed();
+
+   for (NTupleSize_t i = 0; i < kNEvents; i++) {
+      ptrLargeVector->clear();
+      for (std::size_t j = 0; j < kVectorSize; j++)
+         ptrLargeVector->emplace_back(prng->Integer(-1));
+      writer->Fill();
+   }
+   std::cout << "RNTuple written" << std::endl;
+}
+
+/*
+ * ==================================================================================================
+ */
+
+void ReadRNTupleSimple()
+{
+   auto reader = RNTupleReader::Open(kNTupleName, kFileName);
+
+   const auto nEntries = reader->GetNEntries();
+   std::cout << "Simple reading, found " << nEntries << " entries" << std::endl;
+
+   auto ptrLargeVector = reader->GetModel().GetDefaultEntry().GetPtr<std::vector<std::uint32_t>>(kFieldName);
+   for (NTupleSize_t i = 0; i < nEntries; i++) {
+      reader->LoadEntry(i);
+
+      const auto vectorSize = ptrLargeVector->size();
+      uint64_t sum = 0;
+      for (auto val : *ptrLargeVector)
+         sum += val;
+
+      std::cout << "Size and sum of vector: " << vectorSize << " " << sum << std::endl;
+   }
+   std::cout << "RNTuple simple read" << std::endl;
+}
+
+/*
+ * ==================================================================================================
+ */
+
+template <class T>
+class RStreamingVector {
+   RNTupleCollectionView fVectorView;
+   RNTupleView<T> fItemView;
+   RNTupleClusterRange fRange;
+   NTupleSize_t fEntry{0};
+   NTupleSize_t fSize{0};
+
+public:
+   class iterator {
+      RNTupleClusterRange::RIterator fIndex;
+      RNTupleView<T> &fView;
```

Comment:
should store a pointer to make the type copyable

Reply:
Should this be copyable though? To me the semantics would be a bit fuzzy (would the copy automatically get the same window of entries loaded in? Also, when you copy a std::vector you do a deep copy, but in this case to my understanding it wouldn't)

Reply:
Doesn't that hint that the class is more of a `RStreamingVectorView` and/or `RVectorStreamingView`?